src/V3LinkDot.cpp

```diff
@@ -2067,6 +2115,24 @@ class LinkDotResolveVisitor final : public VNVisitor {
             refp->user5p(nodep);
         }
     }
+    VSymEnt* getCreateClockingEventSymEnt(AstClocking* clockingp) {
+        if (!clockingp->eventp()) {
+            auto* const eventp = new AstVar{
+                clockingp->fileline(), VVarType::MODULETEMP, clockingp->name(), VFlagChildDType{},
+                new AstBasicDType{clockingp->fileline(), VBasicDTypeKwd::EVENT}};
+            clockingp->eventp(eventp);
+            // Trigger the clocking event in Observed (IEEE 1800-2017 14.13)
+            clockingp->addNextHere(new AstAlwaysObserved{
+                clockingp->fileline(),
+                new AstSenTree{clockingp->fileline(), clockingp->sensesp()->cloneTree(false)},
+                new AstFireEvent{clockingp->fileline(),
+                                 new AstVarRef{clockingp->fileline(), eventp, VAccess::WRITE},
+                                 false}});
+            v3Global.setHasEvents();
+            eventp->user1p(new VSymEnt{m_statep->symsp(), eventp});
```

Comment:
It seems ugly this is done at link time versus later, but there's other similar ugly stuff going on, so I won't insist we move it later.

Reply:
I don't like it too, but I think it's the last place that makes it simple, because of the clocking block name doubling as the clocking event name.